### PR TITLE
Integrate dependencies 2024.3.0

### PR DESCRIPTION
**Pull request by HugoKulesza**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Dependency update integration

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

### CODE REVIEW

File: docs/user_guide/sensitivity.rst [hunk 1]
```diff
@@ -315,7 +315,7 @@ and postcontingency_branch_flow_factor_matrix methods.
     LOAD         -0.5         -0.5
     >>> result.get_sensitivity_matrix('postcontingency', 'NHV1_NHV2_1')
          NHV1_NHV2_1  NHV1_NHV2_2
-    GEN          0.0          0.0
+    GEN          0.0         -0.0
```
Comment (colineplqt) [inline docs/user_guide/sensitivity.rst:318]: This is odd no?

Comment (HugoKulesza) [reply docs/user_guide/sensitivity.rst:318]: yes all the small changes in test values are strange, but particularly this one

File: java/src/main/java/com/powsybl/python/loadflow/LoadFlowCUtils.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
 public final class LoadFlowCUtils {
 
     static final DataframeMapper<LoadFlowProvider, Void> SPECIFIC_PARAMETERS_MAPPER = new DataframeMapperBuilder<LoadFlowProvider, Parameter, Void>()
-            .itemsProvider(LoadFlowProvider::getSpecificParameters)
+            .itemsProvider(provider -> provider.getSpecificParameters())
```
Comment (colineplqt) [inline java/src/main/java/com/powsybl/python/loadflow/LoadFlowCUtils.java:34]: Why did you do this modification?

**Review by colineplqt — APPROVED** [review]
Except for the little changes in values, which I guess is linked to some improvements of OLF, it looks good to me.

**Pull request merged**